src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.Literal.cs

```diff
@@ -0,0 +1,305 @@
+            Span<char> span;
+            if (length > StackallocThreshold)
+            {
+                propertyArray = ArrayPool<char>.Shared.Rent(length);
+                span = propertyArray;
+            }
+            else
+            {
+                // Cannot create a span directly since the span gets exposed outside this method.
+                unsafe
+                {
+                    char* ptr = stackalloc char[length];
+                    span = new Span<char>(ptr, length);
+                }
+            }
+            JsonWriterHelper.EscapeString(ref propertyName, ref span, firstEscapeIndexProp, out int written);
+            propertyName = span.Slice(0, written);
+
+            WriteLiteralByOptions(ref propertyName, ref value);
+
+            if (propertyArray != null)
+                ArrayPool<char>.Shared.Return(propertyArray);
+        }
+
+        private void WriteLiteralEscapeProperty(ref ReadOnlySpan<byte> propertyName, ref ReadOnlySpan<byte> value, int firstEscapeIndexProp)
+        {
+            Debug.Assert(int.MaxValue / MaxExpansionFactorWhileEscaping >= propertyName.Length);
+
+            byte[] propertyArray = null;
+
+            int length = firstEscapeIndexProp + MaxExpansionFactorWhileEscaping * (propertyName.Length - firstEscapeIndexProp);
+            Span<byte> span;
+            if (length > StackallocThreshold)
+            {
+                propertyArray = ArrayPool<byte>.Shared.Rent(length);
+                span = propertyArray;
+            }
+            else
+            {
+                // Cannot create a span directly since the span gets exposed outside this method.
+                unsafe
+                {
+                    byte* ptr = stackalloc byte[length];
+                    span = new Span<byte>(ptr, length);
+                }
+            }
+            JsonWriterHelper.EscapeString(ref propertyName, ref span, firstEscapeIndexProp, out int written);
+            propertyName = span.Slice(0, written);
+
+            WriteLiteralByOptions(ref propertyName, ref value);
+
+            if (propertyArray != null)
+                ArrayPool<byte>.Shared.Return(propertyArray);
+        }
+
+        private void WriteLiteralByOptions(ref ReadOnlySpan<char> propertyName, ref ReadOnlySpan<byte> value)
+        {
+            int idx;
+            if (_writerOptions.Indented)
+            {
+                if (!_writerOptions.SkipValidation)
+                {
+                    ValidateWritingProperty();
+                }
+                idx = WritePropertyNameIndented(ref propertyName);
+            }
+            else
+            {
+                if (!_writerOptions.SkipValidation)
+                {
+                    ValidateWritingProperty();
+                }
+                idx = WritePropertyNameMinimized(ref propertyName);
+            }
+
+            if (value.Length > _buffer.Length - idx)
```

Comment:
This AdvanceAndGrow is `if`, many others are `while`.  They should be consistent.